### PR TITLE
Update Java Language to Java 19

### PR DESCRIPTION
**Pull request by MBoegers**
### Changes
add missing Java 19 Keywords
* permits #3646
* yield for switch expressions #3647

### Checklist
- [ ] Added markup tests 
- [ ] Updated the changelog at `CHANGES.md`

### CODE REVIEW

File: src/languages/java.js [hunk 1]
```diff
@@ -72,7 +72,10 @@ export default function(hljs) {
     'requires',
     'exports',
     'do',
-    'sealed'
+    'sealed',
+    'non-sealed',
+    'when',
+    'yield'
```
Comment (ffontecha080) [inline src/languages/java.js:78]: Goood

**Review by ffontecha080 — APPROVED** [review]

**Comment by joshgoebel** [comment]
Did you test that `non-sealed` actually works?  I'm not sure `-` is in the default `$pattern`... if not you'll need to add a keyword match pattern that allows for `x-x` style keywords.

File: src/languages/java.js [hunk 2]
```diff
@@ -72,7 +72,10 @@ export default function(hljs) {
     'requires',
     'exports',
     'do',
```
Comment (ffontecha080) [inline src/languages/java.js:74]: export

**Review by ffontecha080 — APPROVED** [review]

**Comment by MBoegers** [comment]
Nope not yet. But will! I’ve also forgotten permits as keyword. Will fix and Test this on Monday.
________________________________
Von: ffontecha080 ***@***.***>
Gesendet: Saturday, October 22, 2022 5:51:14 PM
An: highlightjs/highlight.js ***@***.***>
Cc: Merlin Bögershausen ***@***.***>; Author ***@***.***>
Betreff: Re: [highlightjs/highlight.js] Update Java Language to Java 19 (PR #3641)


@ffontecha080 approved this pull request.

________________________________

In src/languages/java.js<https://github.com/highlightjs/highlight.js/pull/3641#discussion_r1002513604>:

> @@ -72,7 +72,10 @@ export default function(hljs) {
     'requires',
     'exports',
     'do',


export

File: src/languages/java.js (hunk 2 above)
Comment (MBoegers) [reply src/languages/java.js:74]: if you mean line 73, than no 😉. JPMS uses ´exports´ to define a package as available from other modules. See [JSL 7.7. Module Declarations](https://docs.oracle.com/javase/specs/jls/se19/html/jls-7.html#jls-7.7) and [JSL 3.9. Keywords](https://docs.oracle.com/javase/specs/jls/se19/html/jls-3.html#jls-ContextualKeyword)

**Comment by MBoegers** [comment]
> Did you test that `non-sealed` actually works? I'm not sure `-` is in the default `$pattern`... if not you'll need to add a keyword match pattern that allows for `x-x` style keywords.

TBH non-sealed already was part of hightlight.js prio. I'll did a in depth analysis (see linked Tickets). Should we close ad forget this one and start with clear new PRs?

**Comment by joshgoebel** [comment]
> Should we close ad forget this one and start with clear new PRs?

Whatever is easiest for you...

**Comment by MBoegers** [comment]
is fixed in 2 new clean PRs. see #3649 #3650

**Pull request closed without merging**